agent/api/task/task.go

```diff
@@ -709,6 +734,92 @@ func (task *Task) getAllASMSecretRequirements() map[string]apicontainer.Secret {
 	return reqs
 }
 
+// hasLogRouter returns true if the task has a log router container.
+func (task *Task) hasLogRouter() bool {
+	for _, container := range task.Containers {
+		if container.LogRouter != nil { // This is a log router container.
+			return true
+		}
+	}
+	return false
+}
+
+// initializeLogRouterResource initializes the log router task resource and adds it as a dependency of the
+// log router container.
+func (task *Task) initializeLogRouterResource(config *config.Config, resourceFields *taskresource.ResourceFields) error {
+	containerToLogOptions, err := task.collectLogRouterOptions()
+	if err != nil {
+		return errors.Wrap(err, "unable to initialize log router resource")
+	}
+
+	var logRouterResource *logrouter.LogRouterResource
+	for _, container := range task.Containers {
+		logRouter := container.LogRouter
+		if logRouter != nil {
+			var ec2InstanceID string
+			if container.Environment != nil && container.Environment[awsExecutionEnvKey] == ec2ExecutionEnv {
+				ec2InstanceID = resourceFields.EC2InstanceID
+			}
+
+			logRouterResource = logrouter.NewLogRouterResource(config.Cluster, task.Arn, task.Family+":"+task.Version,
+				ec2InstanceID, config.DataDir, logRouter.Type, logRouter.EnableECSLogMetaData, containerToLogOptions)
+			task.AddResource(logrouter.ResourceName, logRouterResource)
+			container.BuildResourceDependency(logRouterResource.GetName(), resourcestatus.ResourceCreated,
+				apicontainerstatus.ContainerCreated)
+			return nil
```

Comment:
It directly returns nil here. Does this mean that it only has one log router container per task?

Reply:
yes, there can only be one.